src/openforms/authentication/contrib/generic_oidc/plugin.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+from typing import ClassVar, Protocol, TypedDict
+
+from django.http import (
+    HttpRequest,
+    HttpResponseBadRequest,
+    HttpResponseBase,
+    HttpResponseRedirect,
+)
+
+from digid_eherkenning.oidc.models import BaseConfig
+from mozilla_django_oidc_db.utils import do_op_logout
+from mozilla_django_oidc_db.views import _RETURN_URL_SESSION_KEY
+
+from openforms.authentication.typing import FormAuth
+from openforms.forms.models import Form
+from openforms.typing import StrOrPromise
+from openforms.utils.urls import reverse_plus
+
+from ...base import BasePlugin, CosignSlice
+from ...constants import CO_SIGN_PARAMETER, FORM_AUTH_SESSION_KEY, AuthAttribute
+from ...exceptions import InvalidCoSignData
+from ...registry import register
+from .constants import OIDC_ID_TOKEN_SESSION_KEY
+
+
+def get_config_to_plugin() -> dict:
+    """
+    Get the mapping of config class to plugin identifier from the registry.
+    """
+    return {
+        plugin.config_class: plugin
+        for plugin in register
+        if isinstance(plugin, OIDCAuthentication)
+    }
+
+
+class AuthInit(Protocol):
+    def __call__(
+        self, request: HttpRequest, return_url: str, *args, **kwargs
```

Comment:
let's make this `options` kwarg explicit! Took me a long time to figure out where the `options` were passed along :sweat_smile: